flow/layers/texture_layer.cc

```diff
@@ -32,6 +32,7 @@ void TextureLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
 
 void TextureLayer::Paint(PaintContext& context) const {
   TRACE_EVENT0("flutter", "TextureLayer::Paint");
+  // FML_DCHECK(needs_painting(context));
```

Comment:
Remove the line that's commented out.

Reply:
Read above for my history on this. Note that the line wasn't actually there before. I added it when I started, realized that the tests were broken and commented it out pending investigation. I've now figured out why it was causing problems and working on fixing it rather than removing it.

Reply:
I've now fixed the tests (new commits pending) and will restore this line.

Reply:
Fixes now committed.